ember/app/components/password-reset-subpanel.js

```diff
@@ -0,0 +1,52 @@
+import Component from '@ember/component';
+import config from 'massbuilds/config/environment';
+import { action, computed } from 'ember-decorators/object';
+import { service } from 'ember-decorators/service';
+
+export default class extends Component {
+
+  @service ajax
+  @service notifications
+
+  constructor() {
+    super();
+
+    this.classNames = ['component', 'subpanel', 'form-subpanel', 'password-reset-subpanel'];
+
+    this.username = '';
+
+    this.errorMessage = null;
+    this.isResetting = false;
+    this.loadingText = 'Resetting Password';
+  }
+
+  @computed('username')
+  get submittable() {
+    return (this.get('username') !== '');
+  }
+
+  @action
+  resetPassword() {
+    if (this.get('submittable') && !this.get('isResetting')) {
+      this.set('isResetting', true);
+      this.set('errorMessage', null);
+
+      const email = this.get('username');
+
+      this.get('ajax').post(`${config.host}/password_resets`, {
+        body: { email },
+      })
+      .then(() => {
+        this.get('notifications').show(`You have successfully reset your password for ${email}! Please check your email.`);
+        this.sendAction('redirect');
+      })
+      .catch(() => {
+        this.set('errorMessage', 'We could not reset your password at this time.');
+      })
+      .finally(() => {
+        this.set('isResetting', false);
+      });
+    }
```

Comment:
This seem pretty strait forward. However, since reset password is mainly a task based function I might recommend either ember-concurrency or using a PromiseProxyMixin. That way you have a lot more freedom and visibility into the state of this task. For example showing a loading spinner, or encapsulating the error message into an object instead of muddying the state of the component.

Reply:
Yeah, I hear you on this. I've created an issue to make note that I'm going to implement ember-concurrency on a future release (#64). This is definitely another area that will benefit, especially considering this pattern is used elsewhere in the app.